addons/mail/controllers/discuss.py

```diff
@@ -400,18 +401,35 @@ def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
         if 'attachments' in request_list:
             res['attachments'] = thread.env['ir.attachment'].search([('res_id', '=', thread.id), ('res_model', '=', thread._name)], order='id desc')._attachment_format(commands=True)
         if 'followers' in request_list:
-            res['followers'] = [{
-                'id': follower.id,
-                'partner_id': follower.partner_id.id,
-                'name': follower.name,
-                'display_name': follower.display_name,
-                'email': follower.email,
-                'is_active': follower.is_active,
-                # When editing the followers, the "pencil" icon that leads to the edition of subtypes
-                # should be always be displayed and not only when "debug" mode is activated.
-                'is_editable': True,
-                'partner': follower.partner_id.mail_partner_format()[follower.partner_id],
-            } for follower in thread.message_follower_ids]
+            res['followers'] = []
+            has_write_access = False
+            try:
+                thread.check_access_rights("write")
+                thread.check_access_rule("write")
+                has_write_access = True
+            except AccessError:
+                pass
+            for follower in thread.message_follower_ids:
+                is_editable = has_write_access
+                if follower.partner_id == request.env.user.partner_id:
+                    # always allow edition of subtypes for logged in user's partner
+                    is_editable = True
+                res['followers'].append({
+                    'id': follower.id,
+                    'partner_id': follower.partner_id.id,
+                    'name': follower.name,
+                    'display_name': follower.display_name,
+                    'email': follower.email,
+                    'is_active': follower.is_active,
+                    # When editing the followers, the "pencil" icon that leads to the edition of subtypes
+                    # should be always be displayed and not only when "debug" mode is activated. Note that
+                    # if the user does not have write access for thread, only own subscription can be
+                    # updated. So in that case, the icon will be displayed only besides logged in user's
+                    # follower whereas it will be hidden for other followers.
+                    'is_editable': is_editable,
+                    'partner': follower.partner_id.mail_partner_format()[follower.partner_id]
+                })
```

Comment:
It should be possible to avoid all of this diff by computing `isEditable` in js directly (and remove `is_editable` from here), as long as `hasWriteAccess` is returned.